src/hooks/swap/useSwapInfo.tsx

```diff
@@ -35,39 +35,37 @@ interface SwapInfo {
 
 // from the current swap inputs, compute the best trade and return it.
 function useComputeSwapInfo(routerUrl?: string): SwapInfo {
-  const isWrap = useIsWrap()
-  const { type, amount, [Field.INPUT]: currencyIn, [Field.OUTPUT]: currencyOut } = useAtomValue(swapAtom)
+  const { type, amount, [Field.INPUT]: inputCurrency, [Field.OUTPUT]: outputCurrency } = useAtomValue(swapAtom)
 
   const parsedAmount = useMemo(
-    () => tryParseCurrencyAmount(amount, (isExactInput(type) ? currencyIn : currencyOut) ?? undefined),
-    [amount, type, currencyIn, currencyOut]
+    () => tryParseCurrencyAmount(amount, (isExactInput(type) ? inputCurrency : outputCurrency) ?? undefined),
+    [amount, inputCurrency, outputCurrency, type]
   )
-  const hasAmounts = currencyIn && currencyOut && parsedAmount && !isWrap
   const trade = useRouterTrade(
     type,
-    hasAmounts ? parsedAmount : undefined,
-    hasAmounts ? (isExactInput(type) ? currencyOut : currencyIn) : undefined,
-    RouterPreference.TRADE,
+    parsedAmount,
+    isExactInput(type) ? outputCurrency : inputCurrency,
+    useIsWrap() ? RouterPreference.SKIP : RouterPreference.TRADE,
     routerUrl
   )
 
-  const amountIn = useMemo(
-    () => (isWrap || isExactInput(type) ? parsedAmount : trade.trade?.inputAmount),
-    [isWrap, parsedAmount, trade.trade?.inputAmount, type]
+  const inputAmount = useMemo(
+    () => (isExactInput(type) ? parsedAmount : trade.trade?.inputAmount),
+    [parsedAmount, trade.trade?.inputAmount, type]
   )
-  const amountOut = useMemo(
-    () => (isWrap || !isExactInput(type) ? parsedAmount : trade.trade?.outputAmount),
-    [isWrap, parsedAmount, trade.trade?.outputAmount, type]
+  const outputAmount = useMemo(
```

Comment:
nit: amountIn/amountOut have protocol meanings that are lost by this change. The memo calls are specifically outputting the protocol's (exact)amount(in/out), so I think these names should stay the same.